sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/_evaluators/_task_adherence/_task_adherence.py

```diff
@@ -0,0 +1,148 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+import os
+import math
+from typing import Dict, Union, List, Optional
+
+from typing_extensions import overload, override
+
+from azure.ai.evaluation._exceptions import EvaluationException, ErrorBlame, ErrorCategory, ErrorTarget
+from azure.ai.evaluation._evaluators._common import PromptyEvaluatorBase
+from azure.ai.evaluation._common.utils import parse_quality_evaluator_reason_score
+from azure.ai.evaluation._model_configurations import Message
+from azure.ai.evaluation._common._experimental import experimental
+
+@experimental
+class TaskAdherenceEvaluator(PromptyEvaluatorBase[Union[str, float]]):
+    """The Task Adherence evaluator assesses how well an AI-generated response follows the assigned task based on:
+
+        - Alignment with instructions and definitions
+        - Accuracy and clarity of the response
+        - Proper use of provided tool definitions
+
+    Scoring is based on five levels:
+    1. Fully Inadherent - Response completely ignores instructions.
+    2. Barely Adherent - Partial alignment with critical gaps.
+    3. Moderately Adherent - Meets core requirements but lacks precision.
+    4. Mostly Adherent - Clear and accurate with minor issues.
+    5. Fully Adherent - Flawless adherence to instructions.
+
+    The evaluation includes a step-by-step reasoning process, a brief explanation, and a final integer score.
+
+
+    :param model_config: Configuration for the Azure OpenAI model.
+    :type model_config: Union[~azure.ai.evaluation.AzureOpenAIModelConfiguration,
+        ~azure.ai.evaluation.OpenAIModelConfiguration]
+
+    .. admonition:: Example:
+        .. literalinclude:: ../samples/evaluation_samples_evaluate.py
+            :start-after: [START task_adherence_evaluator]
+            :end-before: [END task_adherence_evaluator]
+            :language: python
+            :dedent: 8
+            :caption: Initialize and call an TaskAdherenceEvaluator with a query and response.
+    """
+
+    _PROMPTY_FILE = "task_adherence.prompty"
+    _RESULT_KEY = "task_adherence"
+    _OPTIONAL_PARAMS = ["tool_definitions"]
+
+    DEFAULT_TASK_ADHERENCE_SCORE = 5
```

Comment:
use regular defaults instead of max
```suggestion
    DEFAULT_TASK_ADHERENCE_THRESHOLD= 3
```